src/controllers/badgeController.js

```diff
@@ -49,19 +49,25 @@ const badgeController = function (Badge) {
    * - Refactored data validation for duplicate badge id.
    * - Added data validation for badge count should greater than 0.
    * - Added logic to combine duplicate badges into one with updated properties.
+   *
+   * Updated Date: 04/05/2024
+   * Updated By: Abi
+   * Function added:
+   * - Refactored method to utilize async await syntax to make the code more testable.
    */
 
   const assignBadges = async function (req, res) {
     if (!(await helper.hasPermission(req.body.requestor, 'assignBadges'))) {
-      res.status(403).send('You are not authorized to assign badges.');
+      res.status(403).send({ error: 'You are not authorized to assign badges.' });
```

Comment:
You are changing the source, are you sure this won't break the interaction with the frontend?